core-bundle/src/Resources/contao/models/PageModel.php

```diff
@@ -1268,17 +1271,17 @@ public function loadDetails()
 				}
 			}
 
-			if (System::getContainer()->getParameter('contao.legacy_routing'))
+			if ($container->getParameter('contao.legacy_routing'))
 			{
-				$this->urlPrefix = System::getContainer()->getParameter('contao.prepend_locale') ? LocaleUtil::formatAsLanguageTag($objParentPage->language) : '';
-				$this->urlSuffix = System::getContainer()->getParameter('contao.url_suffix');
+				$this->urlPrefix = $container->getParameter('contao.prepend_locale') ? LocaleUtil::formatAsLanguageTag($objParentPage->language) : '';
+				$this->urlSuffix = $container->getParameter('contao.url_suffix');
 			}
 		}
 
 		// No root page found
-		elseif (TL_MODE == 'FE' && $this->type != 'root')
+		elseif ($request && $container->get('contao.routing.scope_matcher')->isFrontendRequest($request) && $this->type != 'root')
```

Comment:
_Note:_ for the tests to work more easily (without workarounds) I got rid of the `TL_MODE` constant here.